src/routes.js

```diff
@@ -494,7 +494,28 @@ const routes = [
       }
     }
   }),
-
+  asyncRoute({
+    path: '/projects/:projectId([1-9]\\d*)/datasets/:datasetName',
+    component: 'DatasetShow',
+    props: true,
+    loading: 'page',
+    key: ({ projectId, datasetName }) =>
+      `/projects/${projectId}/datasets/${encodeURIComponent(datasetName)}`,
+    children: [
+      asyncRoute({
+        path: '',
+        component: 'DatasetOverview',
+        props: true,
+        loading: 'tab',
+        meta: {
+          title: () => [dataset.name],
+          validateData: {
+            project: () => project.permits('dataset.list')
```

Comment:
Shouldn't the verb be `dataset.read` instead of `dataset.list`? You can add new verbs in test/data/seed.js.

Reply:
need to add `dataset.read` in backend before changing here, will create separate PR for both repositories

Reply:
Should we also check for `form.list` in that endpoint since it returns forms? It probably doesn't matter that much, since right now anyone who can `dataset.list` can also `form.list`. But I have this comment in Backend in mind: https://github.com/getodk/central-backend/blob/f5ccb2611c2ae283df458943aef9ef8bb1680742/lib/resources/forms.js#L135-L150